### PR TITLE
NFT Transfer Emit Event and Event Handler

### PR DESCRIPTION
**Pull request by hariria**
#9898

## Motivation

Creating an EventHandler and emitting events for NFT Transfer

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

## Test Plan

```
move package build && cargo x build
```
```
cargo x test
```
passes all except for 
```
cargo test --test move_verification_test
```
fails. I tried running this test on main with clean changes and it failed on there too so I don't think that this is being caused by my changes...

### CODE REVIEW

File: diem-move/diem-framework/experimental/sources/NFT.move [hunk 1]
```diff
@@ -48,8 +48,18 @@ address 0x1 {
             amount: u64,
         }
 
+        struct TransferEvent has copy, drop, store {
+            id: GUID::ID,
+            from: address,
+            to: address,
```
Comment (gregnazario) [inline diem-move/diem-framework/experimental/sources/NFT.move:54]: I'm not really a fan of `from` and `to`, but it looks like that's what we're already using.  I'd prefer sender and receiver, but you can leave it like this for now.

File: diem-move/diem-framework/experimental/sources/NFT.move [hunk 2]
```diff
@@ -48,8 +48,18 @@ address 0x1 {
...
+            amount: u64,
```
Comment (gregnazario) [inline diem-move/diem-framework/experimental/sources/NFT.move:55]: This we might have to think about later if order matters, but we don't have a unique identifier for each individual token (within a GUID).

File: diem-move/diem-framework/experimental/sources/NFT.move [hunk 3]
```diff
@@ -48,8 +48,18 @@ address 0x1 {
...
+            creator: address,
+            creation_num: u64
```
Comment (gregnazario) [inline diem-move/diem-framework/experimental/sources/NFT.move:57]: You don't need these, they're in the `GUID`.  You can drop them.

**Review by gregnazario — COMMENTED** [review]
Looks good, just drop the duplicate fields

**Review by gregnazario — APPROVED** [review]

File: diem-move/diem-framework/experimental/sources/NFT.move (hunk 2 above)
Comment (dmitri-perelman) [reply diem-move/diem-framework/experimental/sources/NFT.move:55]: I hope it's ok: the whole idea of multi-token NFTs is that they're supposed to be fungible within the same GUID.

**Comment by dmitri-perelman** [comment]
@hariria how did you test the newly added events?

**Comment by hariria** [comment]
@dmitri-perelman 
Screen Shot 2021-12-06 at 1 18 01 PM

How I did it:
1. On the terminal, I started a local node with admin module genesis as shown on [line 10 here on demo_commands.sh](https://github.com/diem/diem/commit/9e320a1e4bbcf8936ae5b12a99bc97745de26df4). The admin module holds the event_handlers for `TransferEvents` and `MintEvents`
```bash
cargo run --bin diem-node -- --lazy --test --open-publishing --genesis-modules diem-move/diem-framework/experimental/releases/artifacts/current --seed 0000000000000000000000000000000000000000000000000000000000000000
```
2. Ran the rest of the `demo_commands.sh` script
3. Then curled the transfer event 
```bash
curl http://0.0.0.0:8080/accounts/0xA550C18/events/0x1::NFT::Admin/transfer_events
```

This ultimately returned the transfer event:
```
[{"key":"0x13000000000000000000000000000000000000000a550c18","sequence_number":"0","type":"0x1::NFT::TransferEvent","data":{"amount":"10","from":"0x3132e2b5216a46dfcf8154079954c129ah
```

**Review by gregnazario — APPROVED** [review]

**Comment by dmitri-perelman** [comment]
Nice, thanks!

**Pull request closed without merging**